source/core/slang-char-util.h

```diff
@@ -0,0 +1,51 @@
+#ifndef SLANG_CORE_CHAR_UTIL_H
+#define SLANG_CORE_CHAR_UTIL_H
+
+#include "slang-string.h"
+
+namespace Slang {
+
+struct CharUtil
+{
+    typedef uint8_t Flags;
+    struct Flag
+    {
+        enum Enum : Flags
+        {
+            Upper           = 0x01,
+            Lower           = 0x02,
+            Digit           = 0x04,         
+            LineWhitespace  = 0x08,         ///< Whitespace that can appear in a line (ie excluding CR/LF)
+            HexDigit        = 0x10,         ///< 0-9, a-f, A-F
+        };
+    };
+
+    SLANG_FORCE_INLINE static bool isDigit(char c) { return c >= '0' && c <= '9'; }
+    SLANG_FORCE_INLINE static bool isLower(char c) { return c >= 'a' && c <= 'z'; }
+    SLANG_FORCE_INLINE static bool isUpper(char c) { return c >= 'A' && c <= 'Z'; }
+    SLANG_FORCE_INLINE static bool isLineWhitespace(char c) { return c == ' ' || c == '\t'; }
```

Comment:
Should this be `HorizontalWhitespace`? That's what I assume "line whitespace" means here, and from my googling I don't see the term "line whitespace" used much other than as part of "end-of-line whitespace."